src/renderer/components/ChatList.js

```diff
@@ -110,56 +110,37 @@ class ChatList extends React.Component {
           { !chatList.length && (<ChatListNoChats><p>{missingChatsMsg}</p></ChatListNoChats>) }
           <div className='ConversationList' ref={this.chatListDiv}>
             {chatList.map((chatListItem, i) => {
+              
               if (!chatListItem) return
-              const lastUpdated = chatListItem.summary.timestamp ? chatListItem.summary.timestamp * 1000 : null
+              
 
               // Don't show freshMessageCounter on selected chat
               if (chatListItem.deaddrop) {
                 const name = `${tx('new_message_from_desktop')} ${chatListItem.name}`
                 return (
-                  <ContactRequestItemWrapper key={i}>
+                  <ContactRequestItemWrapper key={chatListItem.id}>
                     <ChatListItem
                       className='contactrequest'
-                      name={name}
-                      lastUpdated={lastUpdated}
-                      lastMessage={{
-                        text1: chatListItem.summary.text1,
-                        text2: chatListItem.summary.text2,
-                        status: 'delivered'
-                      }}
+                      chatListItem={chatListItem}
                       onClick={() => onDeadDropClick(chatListItem.deaddrop)}
                       isSelected={chatListItem.id === selectedChatId}
-                      unreadCount={chatListItem.freshMessageCounter}
                     />
                   </ContactRequestItemWrapper>)
               } else if (chatListItem.isArchiveLink) {
                 return (
                   <ArchivedChats key={chatListItem.id}>
                     <ChatListItem
                       onClick={this.props.onShowArchivedChats}
-                      name={chatListItem.name}
+                      chatListItem={chatListItem}
                     />
                   </ArchivedChats>
                 )
               } else {
                 return (
                   <ChatListItem
-                    key={chatListItem.id}
+                    chatListItem={chatListItem}
```

Comment:
why did you remove key attribute here? It might be an performance issue when reordering the chat list or when inserting new items at the beginning if react has no key to easy "identify" unchanged items. https://reactjs.org/docs/reconciliation.html#recursing-on-children
I know we have an issue with duplicate keys sometimes but that should not be solved by removing the key. Consider that the chat list might have hundreds of items...

Reply:
That wasn't on purpose. I actually added the correct ones to the map values (ContactRequestItemWrapper & ArchivedChats) which fixes the multiple error problem (we had the index of the element in the array + the chatId which in some cases where the same). So this problem is fixed with the latest commit.